Structure Manager - Not possible to delete a Text/Decimal field (MySQL)

Reproduced on a 2.5.6/MySQL instance. Not happening on demo.dotcms.com.

How to reproduce:

1. Go to Structure Manager
2. Edit any structure
3. Add a Text/Decimal field
4. Try to delete the field that you just created

![screen shot 2014-07-30 at 10 20 38 pm](https://cloud.githubusercontent.com/assets/1569509/3759216/76533a0e-1859-11e4-88b1-0c27bd0ce18b.png)


[30/07/14 22:14:07:145 EDT] ERROR action.EditFieldAction: Falied trying to delete field.
com.dotmarketing.exception.DotDataException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '"float1" = 0.0 where structure_inode = '4c441ada-944a-43af-a653-9bb4f3f0cb2b'' at line 1
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:234)
	at com.dotcms.content.elasticsearch.business.ESContentFactoryImpl.cleanField(ESContentFactoryImpl.java:158)
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.cleanField(ESContentletAPIImpl.java:681)
	at com.dotmarketing.portlets.contentlet.business.ContentletAPIInterceptor.cleanField(ContentletAPIInterceptor.java:382)
	at com.dotmarketing.portlets.structure.action.EditFieldAction._deleteField(EditFieldAction.java:521)
	at com.dotmarketing.portlets.structure.action.EditFieldAction.processAction(EditFieldAction.java:156)
	at com.liferay.portal.struts.PortletRequestProcessor.process(PortletRequestProcessor.java:128)
	at com.liferay.portlet.StrutsPortlet.processAction(StrutsPortlet.java:116)
	at com.liferay.portlet.CachePortlet.processAction(CachePortlet.java:130)
	at com.liferay.portal.action.LayoutAction._processPortletRequest(LayoutAction.java:232)
	at com.liferay.portal.action.LayoutAction._processActionRequest(LayoutAction.java:249)
	at com.liferay.portal.action.LayoutAction.execute(LayoutAction.java:93)
	at org.apache.struts.action.RequestProcessor.processActionPerform(RequestProcessor.java:421)
	at org.apache.struts.action.RequestProcessor.process(RequestProcessor.java:226)
	at com.liferay.portal.struts.PortalRequestProcessor.process(PortalRequestProcessor.java:158)
	at org.apache.struts.action.ActionServlet.process(ActionServlet.java:1164)
	at org.apache.struts.action.ActionServlet.doGet(ActionServlet.java:397)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:617)
	at com.liferay.portal.servlet.MainServlet.callParentService(MainServlet.java:275)
	at com.liferay.portal.servlet.MainServlet.service(MainServlet.java:501)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at com.liferay.filters.secure.SecureFilter.doFilter(SecureFilter.java:135)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:404)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:126)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at com.dotmarketing.filters.AutoLoginFilter.doFilter(AutoLoginFilter.java:61)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at com.dotmarketing.cms.urlmap.filters.URLMapFilter.doFilter(URLMapFilter.java:85)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:162)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:146)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:233)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:465)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:555)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:298)
	at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:852)
	at org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:588)
	at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:489)
	at java.lang.Thread.run(Thread.java:744)
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '"float1" = 0.0 where structure_inode = '4c441ada-944a-43af-a653-9bb4f3f0cb2b'' at line 1
	at sun.reflect.GeneratedConstructorAccessor77.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:411)
	at com.mysql.jdbc.Util.getInstance(Util.java:386)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1053)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:4120)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:4052)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2503)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2664)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2794)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2155)
	at com.mysql.jdbc.PreparedStatement.execute(PreparedStatement.java:1379)
	at org.apache.tomcat.dbcp.dbcp.DelegatingPreparedStatement.execute(DelegatingPreparedStatement.java:169)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:564)
	at com.dotmarketing.common.db.DotConnect.executeQuery(DotConnect.java:486)
	at com.dotmarketing.common.db.DotConnect.loadResult(DotConnect.java:232)
	... 55 more

--- a/src/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java
+++ b/src/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java
@@ -130,7 +130,10 @@ public class ESContentFactoryImpl extends ContentletFactory {
 		}*/
 	    StringBuffer sql = new StringBuffer("update contentlet set " );
         if(field.getFieldContentlet().indexOf("float") != -1){
-            sql.append("\""+field.getFieldContentlet()+"\"" + " = ");
+        	if(DbConnectionFactory.isMySql())
+       		 	sql.append(field.getFieldContentlet() + " = ");
+       	    else
+       	    	sql.append("\""+field.getFieldContentlet()+"\"" + " = "); 
         }else{
             sql.append(field.getFieldContentlet() + " = ");
         }

